tests/src/whisk/core/controller/test/AuthenticateTests.scala

```diff
@@ -146,3 +157,55 @@ class AuthenticateTests extends ControllerTestCommon with Authenticate {
         user should be(None)
     }
 }
+
+class AuthenticatedRouteTests
+    extends ControllerTestCommon
+    with Authenticate
+    with AuthenticatedRoute
+    with TransactionCounter {
+
+    behavior of "Authenticated Route"
+
+    val route = sealRoute {
+        implicit val tid = transid()
+        handleRejections(BasicHttpService.customRejectionHandler) {
+            path("secured") {
+                authenticate(basicauth) {
+                    user => complete("ok")
+                }
+            }
+
+        }
+    }
+
+    it should "authorize a known user" in {
+        val creds = createTempCredentials(transid())._1
+        val validCredentials = BasicHttpCredentials(creds.uuid(), creds.key())
+        Get("/secured") ~> addCredentials(validCredentials) ~> route ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "not authorize an unknown user" in {
+        val invalidCredentials = BasicHttpCredentials(UUID().toString, Secret().toString)
+        Get("/secured") ~> addCredentials(invalidCredentials) ~> route ~> check {
+            status should be(Unauthorized)
+        }
+    }
+
+    it should "report service unavailable when db lookup fails" in {
+        implicit val tid = transid()
+        val creds = createTempCredentials._1
+
+        // force another key for the same uuid to cause an internal violation
+        val secondCreds = WhiskAuth(Subject(), AuthKey(creds.uuid, Secret()))
+        put(authStore, secondCreds)
+        waitOnView(authStore, creds.uuid, 2)
```

Comment:
Yes because we aren't setup to do the latter easily. Having a proxy db for testing is a highly desirarable.